exporter/awsemfexporter/internal/appsignals/useragent.go

```diff
@@ -0,0 +1,116 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package appsignals // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/awsemfexporter/internal/appsignals"
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/jellydator/ttlcache/v3"
+	semconv "go.opentelemetry.io/collector/semconv/v1.18.0"
+)
+
+const (
+	handlerName = "aws.appsignals.UserAgentHandler"
+	// defaultTTL is how long an item in the cache will remain if it has not been re-seen.
+	defaultTTL = time.Minute
+	// cacheSize is the maximum number of unique telemetry SDK languages that can be stored before one will be evicted.
+	cacheSize = 5
+	// attrLengthLimit is the maximum length of the language and version that will be used for the user agent.
+	attrLengthLimit = 20
+
+	// TODO: Available in semconv/v1.21.0+. Replace after collector dependency is v0.91.0+.
+	attributeTelemetryDistroVersion = "telemetry.distro.version"
```

Comment:
It looks like we should do remove this now that it's available and no longer relying on a past collector-contrib release.